### PR TITLE
feat: add planetodetic geometry module

### PR DESCRIPTION
**Pull request by phc1990**
Adds planetodetic coordinate system based geometry module.

The idea is to add **sufficient support to check if a satellite is flying over a region of interest**. Other features, such as distances, areas, etc. can be added later on without breaking compatibility.

Objects:
- Point
- Ellipsoid
- Polygon (simple for now, i.e. no holes)

Implementation details:
- mirrored other geometry packages
- leverages boost geometry
- constructors kept private and exposed via static construction to prevent future breaking changes

## Summary by CodeRabbit

* **New Features**
  * Added planetodetic geometry: Point, Polygon, Ellipsoid and base Object for geographic coordinates
  * Python bindings exposing planetodetic module (Point, Polygon, Ellipsoid, Object) and factories (sphere/spheroid/WGS84)

* **Bug Fixes / Improvements**
  * Robust validation, normalization and containment logic for polygon/point operations on ellipsoids

* **Tests**
  * Comprehensive C++ and Python unit tests covering construction, edge cases, containment, equality and string formatting

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/Planetodetic/Object/Spheroid.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                    bool: True if the spheroid is a sphere, False otherwise.
+
+                Example:
+                    >>> spheroid = Spheroid.sphere(6378137.0)
+                    >>> spheroid.is_sphere()  # True
+            )doc"
+        )
+
+        .def_static(
+            "from_equatorial_radius_and_flattening",
```
Comment (vishwa2710) [inline bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/Planetodetic/Object/Spheroid.cpp]: Why prefer this over the native constructor which accepts the same values?

File: include/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Spheroid.hpp [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    /// @brief              Constructs a spherical spheroid (f = 0)
+    ///
+    /// @code
+    ///                     Spheroid spheroid = Spheroid::Sphere(6378137.0) ; // Sphere
+    /// @endcode
+    ///
+    /// @param              [in] aRadius A radius
+    /// @return             Spherical spheroid
+
+    static Spheroid Sphere(const Real& aRadius);
```
Comment (vishwa2710) [inline include/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Spheroid.hpp]: Worth adding a WGS84 based constructor?

Also, should this be spheroid or ellipsoid 🤔

File: src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Polygon.cpp [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+    bool contains(const Point& aPoint) const;
+
+   private:
+    BoostSpheroid boostSpheroid_;
+    BoostPolygon boostPolygon_;
+};
+
+Polygon::Impl::Impl(const Spheroid& aSpheroid, const Array<Point>& anOuterRing)
+    : boostSpheroid_(aSpheroid.getEquatorialRadius(), aSpheroid.getPolarRadius()),
```
Comment (vishwa2710) [inline src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Polygon.cpp]: This is not used anywhere, is there a reason we want to store it?

Comment (phc1990) [reply src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Polygon.cpp]: agreed, not needed

File: src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Polygon.cpp [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+    // Ensure the polygon is properly closed and oriented
+    boost::geometry::correct(boostPolygon_);
+
+    // Validate the polygon after correction
+    std::string validityMessage;
+    if (!boost::geometry::is_valid(boostPolygon_, validityMessage))
+    {
+        throw ostk::core::error::RuntimeError("Polygon is ill-defined, details: [{}]", validityMessage);
+    }
+}
```
[the remark this answers is not in the document]
Comment (phc1990) [reply src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Polygon.cpp]: No longer applicable, currently use in the strategy

File: include/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Spheroid.hpp (hunk 2 above)
Comment (phc1990) [reply include/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Spheroid.hpp]: Added WGS84 static constructor ✅ 

> Also, should this be spheroid or ellipsoid

I thought about this before creating it. Technically a planetodetic/planetographic coordinate system would deal with 3-axes ellipsoids, however most practial earth-bound applications and/or libraries (including boost), and even boost only truly supports oblate spheroids.

Having said this, I think it's best to keep the door open. so what I'll do is:

1. Generalise this class `Spheroid` -> `Ellipsoid`
2. Allow easy spheroid creation via static call `::Spheroid(a, f)`
3. Include runtime checks when a function is not supported (just like we're already doing when checking `contains` on prolate spheroids)

File: bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/Planetodetic/Object/Spheroid.cpp (hunk 1 above)
Comment (phc1990) [reply bindings/python/src/OpenSpaceToolkitMathematicsPy/Geometry/Planetodetic/Object/Spheroid.cpp]: Resolving, no longer applicable, see new Ellipsoid class

File: src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Point.cpp [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    const Vector3d b = aPoint1.toVector();
+    const Vector3d c = aPoint2.toVector();
+
+    // Normal of plane through A and B
+    const Vector3d n = a.cross(b);
+
+    // Dot product of normal with C
+    const double dotNC = n.dot(c);
+
+    return std::abs(dotNC) < 1e-10;
```
Comment (lucas-bremond) [inline src/OpenSpaceToolkit/Mathematics/Geometry/Planetodetic/Object/Point.cpp]: I don't think this math is correct on a curved surface...?
